### PR TITLE
feat: Add LanguageSwitch functionality to toggle between French and English and add a new res/values-fr to take french values

### PR DESCRIPTION
**Pull request by SaFouNaldo**
### PR Description

- **Created a composable named `LanguageSwitch` in `SettingsScreen.kt`** that allows the user to toggle between French (FR) and English (EN) languages.
- **Created a test to cover the lines** ensuring proper functionality and logic of the `LanguageSwitch` composable.
- **Added `updateLanguage` utility** in the `utils` package, which changes the app's resource folders from the default (English) to French (`values-fr`).
- **Updated related classes** in the `SettingsScreen` to adapt to the new language-switching format.

### Future Work

For now, the feature implements only the logic for language switching. Many hardcoded strings in the app still need to be moved into the `strings.xml` files and translated into French (`values-fr`). This will be addressed in a future  [task in the project](https://github.com/orgs/Signify-epfl/projects/2/views/1?filterQuery=resour&pane=issue&itemId=88727837&issue=Signify-epfl%7Csignify-app%7C321). Currently, only the strings defined in `strings.xml` are affected by the language switch.

### Test

- Achieved the desired line coverage of **~81%** with tests validating the language toggle functionality and utility updates.

### CODE REVIEW

File: app/src/main/java/com/github/se/signify/ui/Utils.kt [hunk 1]
```diff
@@ -824,6 +827,30 @@ fun CameraPlaceholder(handLandMarkViewModel: HandLandMarkViewModel, modifier: Mo
...
+ * updateLanguage(context, "en") // Switches the application language to English
+ * *
+ */
+fun updateLanguage(context: Context, languageCode: String) {
+  val locale = Locale(languageCode)
+  Locale.setDefault(locale)
+  val config = Configuration()
+  config.locale = locale
+  context.resources.updateConfiguration(config, context.resources.displayMetrics)
+}
```
Comment (nada-ezziti) [inline app/src/main/java/com/github/se/signify/ui/Utils.kt:853]: The updateLanguage function currently assumes the languageCode provided is valid. I think adding validation for the language code (e.g., checking against a list of supported locales) would make it more robust.
```suggestion
fun updateLanguage(context: Context, languageCode: String) {
if (!listOf("en", "fr").contains(languageCode)) {
    throw IllegalArgumentException("Unsupported language code: $languageCode")
}
  val locale = Locale(languageCode)
  Locale.setDefault(locale)
  val config = Configuration()
  config.locale = locale
  context.resources.updateConfiguration(config, context.resources.displayMetrics)
}
```

**Review by nada-ezziti — CHANGES_REQUESTED** [review]
### **Review Summary: Language Switching Feature**

This PR is a well-executed implementation of a language toggle feature, allowing users to switch between French (FR) and English (EN) on the `SettingsScreen`. The `LanguageSwitch` composable is intuitive and cleanly designed, while the `updateLanguage` utility is well-documented and handles dynamic locale updates effectively. High test coverage (~81%) was achieved, ensuring that the language toggle logic works as intended. 

The integration of SharedPreferences for persisting language preferences is a thoughtful touch, enhancing user experience. This feature provides a solid foundation for multi-language support in the app, with room for future expansion, such as migrating hardcoded strings to `strings.xml` for full localization.

**Required Change:**  
To enhance robustness, I requested adding validation for the language code in the `updateLanguage` utility to handle unsupported languages gracefully:

Everything else is well-implemented, making this a great addition to the project. It’s a strong feature that improves the app's usability and accessibility.

File: app/src/androidTest/java/com/github/se/signify/ui/language/LanguageUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.github.se.signify.ui.updateLanguage
+import java.util.Locale
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+class LanguageUtilsTest {
```
Comment (nada-ezziti) [inline app/src/androidTest/java/com/github/se/signify/ui/language/LanguageUtilsTest.kt:11]: Requested Change : 
Following my previous comment, you can add a test for invalid language codes to ensure that the updateLanguage function handles them correctly.

File: app/src/main/java/com/github/se/signify/ui/Utils.kt (hunk 1 above)
Comment (SaFouNaldo) [reply app/src/main/java/com/github/se/signify/ui/Utils.kt:853]: You are absolutly right, thanks for pointing that (Nice catch ! ).
And thank you for giving the code it's easy to integrate !

File: app/src/androidTest/java/com/github/se/signify/ui/language/LanguageUtilsTest.kt (hunk 2 above)
Comment (SaFouNaldo) [reply app/src/androidTest/java/com/github/se/signify/ui/language/LanguageUtilsTest.kt:11]: Perfect, I should have tested edge cases thanks for pointing that

**Review by nada-ezziti — APPROVED** [review]
Everything seems good now. I approve it.

**Pull request merged**